odl/trafos/wavelet.py

```diff
@@ -653,22 +704,29 @@ def _call(self, x):
             The length of the array depends on the size of input image to
             be transformed and on the chosen wavelet basis.
         """
-        if x.space.ndim == 1:
-            coeff_list = pywt.wavedec(x, self.wbasis, self.mode, self.nscales)
-            coeff_arr = pywt_coeff_to_array(coeff_list, self.size_list)
-            return self.range.element(coeff_arr)
-
-        if x.space.ndim == 2:
-            coeff_list = pywt.wavedec2(x, self.wbasis, self.mode, self.nscales)
-            coeff_arr = pywt_coeff_to_array(coeff_list, self.size_list)
-            return self.range.element(coeff_arr)
-
-        if x.space.ndim == 3:
-            coeff_dict = wavelet_decomposition3d(x, self.wbasis, self.mode,
-                                                 self.nscales)
-            coeff_arr = pywt_coeff_to_array(coeff_dict, self.size_list)
-
-            return self.range.element(coeff_arr)
+        if not isinstance(self.wbasis, pywt.Wavelet):
+            Op = BiorthWaveletTransform(dom=self.domain, nscales=self.nscales,
+                                        wbasis=self.wbasis)
+            coeff = Op(x)
+            return self.range.element(coeff)
```

Comment:
The range of `BiorthWaveletTransform` doesn't need to be called, the range is determined inside the class. Now the user only needs to know the "image domain" and thus the calling of  `BiorthWaveletTransform` is uniform with  calling of `WaveletTransform`.